### PR TITLE
DAC6-464 - Fix routing in the change answers journey

### PR DESCRIPTION
**Pull request by iservport**
This is a bug fix. The route cause is that the calling journey must be remembered by the sub-journey when in check mode. The solution is to wrap the mode in a case class that can be used by the navigator to remember the calling journey.

### CODE REVIEW

File: test/controllers/individual/IndividualAddressControllerSpec.scala [hunk 1]
```diff
@@ -156,8 +151,6 @@ class IndividualAddressControllerSpec extends SpecBase with MockitoSugar with Nu
 
       status(result) mustEqual SEE_OTHER
 
-      redirectLocation(result).value mustEqual onwardRoute.url
```
Comment (helensoniega) [inline test/controllers/individual/IndividualAddressControllerSpec.scala:159]: Add redirect check for new Navigator

File: test/controllers/individual/IsIndividualAddressKnownControllerSpec.scala [hunk 2]
```diff
@@ -129,8 +126,6 @@ class IsIndividualAddressKnownControllerSpec extends SpecBase with MockitoSugar
 
       status(result) mustEqual SEE_OTHER
 
-      redirectLocation(result).value mustEqual onwardRoute.url
```
Comment (helensoniega) [inline test/controllers/individual/IsIndividualAddressKnownControllerSpec.scala:132]: Same here

File: test/controllers/individual/IsIndividualResidentForTaxOtherCountriesControllerSpec.scala [hunk 3]
```diff
@@ -142,8 +137,6 @@ class IsIndividualResidentForTaxOtherCountriesControllerSpec extends SpecBase wi
 
       status(result) mustEqual SEE_OTHER
 
-      redirectLocation(result).value mustEqual onwardRoute.url
```
Comment (helensoniega) [inline test/controllers/individual/IsIndividualResidentForTaxOtherCountriesControllerSpec.scala:145]: Same here

File: test/controllers/individual/WhatAreTheTaxNumbersForNonUKIndividualControllerSpec.scala [hunk 4]
```diff
@@ -148,40 +145,8 @@ class WhatAreTheTaxNumbersForNonUKIndividualControllerSpec extends SpecBase with
       val result = route(application, request).value
 
       status(result) mustEqual SEE_OTHER
-      redirectLocation(result).value mustEqual onwardRoute.url
```
Comment (helensoniega) [inline test/controllers/individual/WhatAreTheTaxNumbersForNonUKIndividualControllerSpec.scala:151]: Same here

**Review by soljeggers — APPROVED** [review]

**Pull request merged**